### ISSUE TITLE
Add reminder and sync it with forgetting curve

### ISSUE THREAD

**Issue by mgolokhov** [mgolokhov/dodroid#33]
https://en.wikipedia.org/wiki/Forgetting_curve
alarms + notifications

**Comment by JaeW** [comment]
Good point!  I would also add that it is good practice to review the information many times, but also to review it during different times of the day and in different environments.  The biochemistry of the brain changes depending on the time of day and/or environment - the brain functions best with recall when the same biochemical environment is present as when the information was learned.  That means repeating the same questions at varying times would be ideal (but also optional) So, I think your alarms & notifications is a fantastic suggestion.  

Here's the question.  How to track this?  I saw in the most recent pull request that you are using a singleton.  Since that's locally resident AND the user's input can be collected there, I assume fields could be added to collect the most recent date a question was answered, and how many times that question was answered correctly/incorrectly.  Yes/no?  But how would you maintain user-specific data if a new/corrected/revised version of the questions is pushed to the user's device?

Create preferences/options so user can choose time frame of alerts, and number of review questions at one time?

**Comment by mgolokhov** [comment]
We should invent question status (on learning, studied, ignore,...) and set timers (for every question) according with state. 
A user should be able to configure some preferences (e.g. preferred interval to be disturbed).

**Comment by mgolokhov** [comment]
Project with the same concept and nice statistics https://github.com/ankidroid/Anki-Android.
